src/plugins/intel_gpu/src/graph/include/pass_manager.h

```diff
@@ -369,6 +370,34 @@ class mark_runtime_skippable_nodes : public base_pass {
 public:
     mark_runtime_skippable_nodes() : base_pass("mark_runtime_skippable_nodes") {}
 
+    bool check_consecutive_runtime_skippable(program_node* node, size_t count) {
+        std::unordered_set<program_node*> visited;
+        return check_consecutive_runtime_skippable(node, count, visited);
+    }
+
+    bool check_consecutive_runtime_skippable(program_node* node, size_t count, std::unordered_set<program_node*>& visited) {
+        if (count == 0)
+            return true;
+
+        if (visited.find(node) != visited.end())
+            return false;
+
+        visited.insert(node);
+
+        if (node->is_runtime_skippable()) {
+            count--;
+        } else {
+            return false;
+        }
+
+        for (const auto& dep : node->get_dependencies()) {
+            if (check_consecutive_runtime_skippable(dep.first, count, visited)) {
```

Comment:
Doing this recursive call for every time seems time consuming. 
How about record runtime_skip_depth for each node? something like 
1. when current node can be set as runtime skippable 
2. check its parents' recorded depth 
3. record the depth of cur node as max(parents' depth) + 1

Reply:
Also please add function implementation in cpp file to avoid future compilation overhead due to the modification

Reply:
Implemented it. Thanks!